src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -292,6 +298,65 @@ public void mergeStagingLoadHistoryTable(Dataset dataset, String flightId)
     bigQueryProject.query(sqlTemplate.render());
   }
 
+  private static final String getLoadHistoryTemplate =
+      "SELECT * "
+          + "FROM `<project>.<dataset>.<loadTable>` L "
+          + "WHERE L.load_tag = @loadTag "
+          + "LIMIT <limit> "
+          + "OFFSET <offset>";
+
+  public List<BulkLoadHistoryModel> getLoadHistory(
+      Dataset dataset, String loadTag, int offset, int limit) {
+    BigQueryProject bigQueryProject = BigQueryProject.from(dataset);
+    String datasetName = prefixName(dataset.getName());
+    var sqlTemplate = new ST(getLoadHistoryTemplate);
+    sqlTemplate.add("project", bigQueryProject.getProjectId());
+    sqlTemplate.add("dataset", datasetName);
+    sqlTemplate.add("loadTable", PDAO_LOAD_HISTORY_TABLE);
+    sqlTemplate.add("limit", limit);
+    sqlTemplate.add("offset", offset);
+
+    var query = sqlTemplate.render();
+    QueryJobConfiguration queryConfig =
+        QueryJobConfiguration.newBuilder(query)
+            .addNamedParameter("loadTag", QueryParameterValue.string(loadTag))
+            .build();
+
+    try {
+      var results = bigQueryProject.getBigQuery().query(queryConfig);
+      return StreamSupport.stream(results.iterateAll().spliterator(), true)
+          .map(BigQueryPdao::bigQueryResultToBulkLoadHistoryModel)
+          .collect(Collectors.toList());
+    } catch (InterruptedException ex) {
+      throw new GoogleResourceException("Could not query BigQuery for load history", ex);
+    }
+  }
+
+  private static BulkLoadHistoryModel bigQueryResultToBulkLoadHistoryModel(
+      FieldValueList fieldValue) {
+    Function<FieldValue, String> emptyToNull =
+        fv -> {
+          var value = fv.getStringValue();
+          return value.isEmpty() ? null : value;
+        };
```

Comment:
It seems odd to me to declare the functional type here when you can declare the method normally and Java will convert it to the functional type for you:
```
  private static String emptyToNull(FieldValue fv) {
    var value = fv.getStringValue();
    return value.isEmpty() ? null : value;
  }

[...]

    Optional.ofNullable(fieldValue.get(LoadHistoryUtil.CHECKSUM_CRC32C_FIELD_NAME))
        .map(BigQueryPdao::emptyToNull).
        .ifPresent(BulkLoadHistoryModel::checksumCRC);
```

But I can see the benefits of doing it this way too. If you want to keep it this way, I would replace the declared type `Function<FieldValue, String>` with `var` and change the optional code to use `map()` to avoid having to call `Function.apply()` explicitly.

Reply:
I'm actually changing this to avoid the optional altogether and reconfiguring to be able to use the builder pattern of `BulkLoadHistoryModel`